sqlmesh/core/plan/builder.py

```diff
@@ -180,9 +181,15 @@ def set_choice(self, snapshot: Snapshot, choice: SnapshotChangeCategory) -> Plan
             snapshot: The target snapshot.
             choice: The user decision on how to version the target snapshot and its children.
         """
-        plan = self.build()
-        self._set_choice(snapshot, choice, plan.directly_modified, plan.indirectly_modified)
-        self._adjust_new_snapshot_intervals()
+        if self._forward_only:
```

Comment:
Do we also need to check that the category is not assigned to an indirectly modified snapshot?

Reply:
i don't understand this comment, can you explain it to me?